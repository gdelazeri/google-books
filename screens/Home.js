import React from 'react';
import {
  Text,
  View,
} from 'react-native';
import Constants from 'expo-constants';
import { SearchBar } from 'react-native-elements';
import Styles from '../constants/Styles';
import Colors from '../constants/Colors';
import Screen from '../components/Screen';

// Tempo de espera para buscar os livros
const WAIT_INTERVAL = 600;

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      refreshing: false,
      loadError: false,
      books: [],
    }
  }

  static navigationOptions = {
    headerStyle: { elevation: 0, shadowOpacity: 0 },
    header: null,
  }

  async componentDidMount() {
    this.load();
  }

  async load(refreshing = false) {
    try {
      this.setState({ loadError: false, refreshing: false, loading: false });
    } catch (error) {
      this.setState({ loadError: true, refreshing: false, loading: false });
    }
  }


  async updateSearch(search) {
    clearTimeout(this.timer);
    this.setState({ search });
    if (search.length > 0) {
      this.timer = setTimeout(async () => {
        // const books = await GoogleBooksAPI.listBooks();
        this.setState({ books, filtering: false });
        Keyboard.dismiss();
      }, WAIT_INTERVAL);
    } else {
      this.setState({ books: [] });
    }
  }

  reload() {
    this.setState({ loading: true });
    this.load();
  }

  render() {
    return <Screen loading={this.state.loading} navigation={this.props.navigation} error={this.state.loadError} reload={() => this.reload()}>
      <View style={{ height: Constants.statusBarHeight }} />
      <Text style={[Styles.padding20, Styles.text24, Styles.textBold]}>Google Books</Text>
      <View style={{ flex: 1 }}>
        <SearchBar
          platform={Platform.OS}
          placeholder="Buscar livros"
          placeholderTextColor={Colors.lightText}
          onChangeText={(search) => this.updateSearch(search)}
          value={this.state.search}
          round={true}
          lightTheme={true}
          containerStyle={{ backgroundColor: 'white' }}
          inputContainerStyle={{ backgroundColor: Colors.background, padding: 0, margin: 0, borderRadius: 10 }}
          containerStyle={
            Platform.OS === 'android' ?
            { marginLeft: 0, marginRight: 0, paddingLeft: 10, paddingRight: 10, backgroundColor: 'white' } :
            { margin: 0, padding: 0, paddingLeft: 5, backgroundColor: 'white' }
          }
          onCancel={() => this.updateSearch('')}
          onClear={() => this.updateSearch('')}
          cancelButtonTitle='Cancelar'
          cancelButtonProps={{ buttonTextStyle: { fontSize: 16 } }}
          inputStyle={{ fontSize: 16, color: Colors.defaultText }}
        />
      </View>
    </Screen>
  }
}
