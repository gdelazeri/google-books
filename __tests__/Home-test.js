import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import Home from '../screens/Home';

describe('Home', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`renders the Home screen`, () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
