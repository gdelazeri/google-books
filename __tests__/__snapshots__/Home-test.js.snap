// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the Home screen 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": 44,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "padding": 20,
        },
        Object {
          "paddingBottom": 15,
        },
        Object {
          "color": "#707d8a",
          "fontFamily": "open-sans",
          "fontSize": 24,
        },
        Object {
          "color": "#707d8a",
          "fontFamily": "open-sans-bold",
        },
        Object {
          "color": "#1a73e8",
        },
      ]
    }
  >
    Google Books
  </Text>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "margin": 0,
          "overflow": "hidden",
          "padding": 0,
          "paddingBottom": 13,
          "paddingLeft": 5,
          "paddingTop": 13,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f5f5f5",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 10,
              "flexDirection": "row",
              "margin": 0,
              "marginLeft": 8,
              "marginRight": 8,
              "minHeight": 36,
              "padding": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            lightTheme={true}
            onBlur={[Function]}
            onCancel={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Buscar livros"
            placeholderTextColor="#707d8a"
            platform="ios"
            rejectResponderTermination={true}
            round={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#707d8a",
                "flex": 1,
                "fontSize": 16,
                "marginLeft": 6,
                "minHeight": 40,
                "overflow": "hidden",
              }
            }
            testID="searchInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": -0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "fontSize": 16,
                  },
                  undefined,
                ]
              }
            >
              Cancelar
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={[Function]}
      ListFooterComponent={
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        />
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      keyboardShouldPersistTaps="handled"
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View>
          <View
            style={
              Object {
                "height": 10,
              }
            }
          />
          <View
            style={
              Object {
                "height": 10,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
